README.md

```diff
@@ -1,9 +1,21 @@
 # Debian dependency package generation
 
-A dependency package represents a desired state of a system, which is defined by a list of Debian packages with corresponding versions. All required packages are captured as dependencies in the control file of an otherwise shallow Debian package. APT will try to automatically resolve and install these dependencies when running `apt install dependency-pack`. APT is very opinionated about which dependent package version to install when resolving dependencies: It will always try to install the latest available package version that can be found in all source package repositories unless instructed differently using [APT preferences](https://manpages.debian.org/bullseye/apt/apt_preferences.5.en.html).
+## Motivation
+
+In device management, it is valuable to have the fleet of devices as consistent as possible. This helps in testing software releases and makes it easier to debug issues on remote devices.
+
+If performing package based updates, one topic to consider is how to avoid drift between the initial device image and the state of the device fleet.
+
+One additional consideration is that APT is very opinionated about which dependent package version to install when resolving dependencies. APT will always try to install the latest available package version that can be found in all source package repositories unless instructed differently using [APT preferences](https://manpages.debian.org/bullseye/apt/apt_preferences.5.en.html).
```

Comment:
Nit: I have a concern regarding this part being here. At this point of the README is not clear that APT will be used, and why resolving dependency in a correct way is important. I think it would be more clear if it goes under `Approach with a dependency package`.

Reply:
I did consider this and here is my rationale.

At this point, the reader has seen the title `Debian dependency package generation`. We could introduce `APT` but was thinking that the reader should have basic understanding of Debian and APT or otherwise this project is not very relevant in the first place.

In the `motivation`, I tried to capture the problem statement we are solving in the project. The APT quirks are part of the problem to which we needed a solution for.

Reply:
Ok, makes sense. Then all good!